undertow/src/main/java/org/wildfly/extension/undertow/Server.java

```diff
@@ -169,8 +174,23 @@ public String getDefaultHost() {
     }
 
     public String getRoute() {
-        UndertowService service = this.undertowService.getValue();
-        String defaultServerRoute = service.getInstanceId();
+        final UndertowService service = this.undertowService.getValue();
+        final String defaultServerRoute = service.getInstanceId();
+        if (service.isObfuscateSessionRoute()) {
+            try {
+                final MessageDigest md = MessageDigest.getInstance("SHA-256");
```

Comment:
Given that we are already salting the hash of the instance-id (usually the host name) using the server name, thus preventing mapping back to known host names, is SHA-256 really necessary here?
This results in route identifiers of 43 characters, as opposed to 22 characters when using MD5.

Perhaps we leave this determination up to the user, i.e. make this configurable?
This would replace obfuscate-session-route="true|false" with session-route-digest-algorithm="...", where undefined means we that do not obfuscate.  WDYT?

Reply:
@fl4via To reiterate, we do not require a 32 byte hash algorithm for this.  MD5 is sufficient for the purpose of obfuscating the session route and is consistent with the requested behavior in the original feature request.
We can discuss making this configurable at a later date.

Reply:
@pferraro I'm updating the PR with this change